text/0000-allow-sharing-of-LWC-across-namespaces.md

```diff
@@ -0,0 +1,230 @@
+---
+title: Allow sharing of LWC across namespaces 
+status: DRAFTED
+created_at: 2020-03-29
+updated_at: 2020-04-21
+pr: (leave this empty until the PR is created)
+---
+
+# Allow sharing of LWC across namespaces
+
+## Summary
+
+There is a tight coupling (1:1) between namespace and Lighting Web Components (LWC). Developer can use isExposed flag to expose the component to the world or keep it private within the namespace.
+
+This RFC proposes functionality to allow sharing of LWC across namespaces.
+
+
+## Motivation
+
+Some teams have more than one namespaces and only desire to share their common components internally. However, it is not possible today, which forces such teams to make their component available globally. This is an undesired effect as it introduces maintenance burden.
+
+## Basic example
+
+**Proposal** - Don't touch IsExposed flag for now
+
+We will introduce new tag, **namespaceScope** in LightingComponentBundle.xml (**-meta.xml) file. It’s a string which can take following value viz. global, targeted or local. This tag can be missing in that case we treat it as MISSING internally. 
+
+
+**Global** : Acts similar to isExposed to true flag. It opens the LWC to the world.
+**Targeted** : It implies that we are opening this LWC to limited namespaces. 
+**Local** : This LWC is only open to local namespace
+
+We will also introduced new xml file called NamespaceBundle.xml. This file is at namespace level. It will enlist namespaces where all the LWCs can be shared. This will allow sharing for LWCs to targeted namespaces. 
+
+For now we will honor isExposed flag and display error for in valid scenarios listed below. However, in the future we will completely rely on namespaceScope tag. 

```

Comment:
Since isExposed is boolean and we have tri states like Global, targed and local plan was to introduce new tag and retired the old one. So for now namespaceScope = global should behave exactly like isExposed=true.